sale_order_general_discount_triple/models/sale_order_line.py

```diff
@@ -4,51 +4,44 @@
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
-    discount = fields.Float(
-        string="Discount (%)",
-        digits="Discount",
-        compute=False,
-        store=True,
-        readonly=False,
-        default=0.0,
-    )
+    discount2 = fields.Float(compute="_compute_discount2", store=True, readonly=False)
+    discount3 = fields.Float(compute="_compute_discount3", store=True, readonly=False)
 
-    @api.model
-    def default_get(self, fields):
-        vals = super().default_get(fields)
-        general_discount = (
-            self.env["ir.config_parameter"]
-            .sudo()
-            .get_param(
-                "sale_order_general_discount_triple.general_discount", "discount"
-            )
-        )
-        value = vals.get("discount") or 0.0
-        vals.update(
-            {
-                "discount": 0.0,
-                "discount2": 0.0,
-                "discount3": 0.0,
-            }
-        )
+    @api.depends("product_id", "product_uom", "product_uom_qty")
+    def _compute_discount(self):
+        if hasattr(super(), "_compute_discount"):
+            super()._compute_discount()
+        general_discount = self._get_general_discount_field()
+        for line in self:
+            vals = {"discount": 0.0}
+            if general_discount == "discount":
+                vals.update({"discount": line.order_id.partner_id.sale_discount})
+            line.update(vals)
+        return
 
-        vals.update(
-            {
-                general_discount: value,
-            }
-        )
-        return vals
+    @api.depends("product_id", "product_uom", "product_uom_qty")
+    def _compute_discount2(self):
+        general_discount = self._get_general_discount_field()
+        for line in self:
+            vals = {"discount2": 0.0}
+            if general_discount == "discount2":
+                vals.update({"discount2": line.order_id.partner_id.sale_discount})
+            line.update(vals)
+
+    @api.depends("product_id", "product_uom", "product_uom_qty")
+    def _compute_discount3(self):
+        general_discount = self._get_general_discount_field()
+        for line in self:
+            vals = {"discount3": 0.0}
+            if general_discount == "discount3":
```

Comment:
Exit early if the discount field is not the one we are looking for, and you can pre-assign `self.discount3 = 0`

Reply:
Done!